djitellopy/tello.py

```diff
@@ -510,34 +513,42 @@ def send_read_command_float(self, command: str) -> float:
         return float(response)
 
     def raise_result_error(self, command: str, response: str) -> bool:
-        raise Exception('Command {} was unsuccessful. Message: {}'.format(command, response))
+        tries = 1 + self.retry_count
+        raise Exception("Command '{}' was unsuccessful for {} tries. Latest response:\n\t'{}'\n"
+            .format(command, tries, response))
 
     def connect(self, wait_for_state=True):
         """Enter SDK mode. Call this before any of the control functions.
         """
         self.send_control_command("command")
 
         if wait_for_state:
-            for i in range(20):
+            REPS = 20
+            for i in range(REPS):
                 if self.get_current_state():
-                    Tello.LOGGER.debug('connect() received first state packet after {} seconds'.format(i * 0.05))
+                    t = i / REPS  # in seconds
+                    Tello.LOGGER.debug("'.connect()' received first state packet after {} seconds".format(t))
                     break
-
-                time.sleep(0.05)
+                self.sleep(1 / REPS)
 
             if not self.get_current_state():
                 raise Exception('Did not receive a state packet from the Tello')
 
+    def sleep(time_sleep=5):
```

Comment:
Should I go with 5 here or nothing or do you have another preference? (5 is just what I needed for me small drone to stabilize between commands, but I've only used the thing for 3 days)